Feature add hackathon

tjclawson opened the pull request:
# Description

Add ability to post hackathon to api through app

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Change Status

- [ ] Complete, tested, ready to review and merge
- [X] Complete, but not tested (may need new tests)
- [ ] Incomplete/work-in-progress, PR is for discussion/feedback

# How Has This Been Tested?

- [ ] Test A
- [ ] Test B

# Checklist

- [ ] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] My code has been reviewed by at least one peer
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] There are no merge conflicts

tjclawson committed:
Add Hackathon, Team, and User models

tjclawson committed:
Add basic xml for AddHackathonFragment, add class for AddHackathonFragment

tjclawson committed:
Add HackathonApiInterface and change BASE_URL to correct url in AppModule

tjclawson committed:
Add provideHackathonService to AppModule

tjclawson committed:
Add repository for api calls, edit postHackathon in HackathonApiInterface to include path parameters, move AddHackathonFragment to own package

tjclawson committed:
Add AddHackathonViewModel and bind method to ViewModelsModule

tjclawson committed:
Add ability to post hackathon

tjclawson committed:
Resolve merge conflicts

tjclawson committed:
Modified userinfo model to hold access token

Co-authored-by: 05nelsonm <mjnelso2@protonmail.com>
Co-authored-by: msmaitran <itsmaitran@gmail.com>

tjclawson committed:
Refactor to use DI

Co-authored-by: 05nelsonm <mjnelso2@protonmail.com>
Co-authored-by: msmaitran <itsmaitran@gmail.com>

tjclawson committed:
Cleaned up imports

Co-authored-by: 05nelsonm <mjnelso2@protonmail.com>
Co-authored-by: msmaitran <itsmaitran@gmail.com>

05nelsonm commented on app/src/main/java/com/lambdaschool/hackathon_portal/HackathonRepository.kt:
```diff
@@ -0,0 +1,52 @@
+                override fun onFailure(call: Call<Hackathon>, t: Throwable) {
+                    addHackathonResponse.value = "Failed to connect to API"
+                    Log.i(REPO_TAG, "Failed to connect to API")
+                    Log.i(REPO_TAG, t.message.toString())
```
Display error to user via Toast?

tjclawson replied:
Implemented in commit 4f768f5f09460f6d65b371a974ff92469d63b2a7

05nelsonm commented on app/src/main/java/com/lambdaschool/hackathon_portal/di/FragmentComponent.kt:
```diff
@@ -26,4 +27,5 @@ interface FragmentComponent {
     fun injectDashboardFragment(fragment: DashboardFragment)
     fun injectAccountFragment(fragment: AccountFragment)
     fun injectSettingsFragment(fragment: SettingsFragment)
+    fun injectAddHackathonFragment(fragmen: AddHackathonFragment)
```
typo in `fragment`

Change to `CreateHackathonFragment` to be more descriptive?

tjclawson replied:
Fixed in commit f4fe16d5d80bccd8f11760acbf005fe5a10b7359

05nelsonm commented on app/src/main/java/com/lambdaschool/hackathon_portal/ui/MainActivity.kt:
```diff
@@ -77,6 +77,10 @@ class MainActivity : AppCompatActivity() {
                 }
+
+                R.id.nav_drawer_add_hackathon -> {
+                    nav_host_fragment.findNavController().navigate(R.id.addHackathonFragment)
```
My oversight on PR #12. Forgot to refactor these calls.

navController is injected on line 40 and can be called directly

```suggestion
                    navController.navigate(R.id.addHackathonFragment)
```

tjclawson replied:
Changed in commit 3b1138a1fc3b243d79ce1c20727eac01c3287d03

05nelsonm commented on app/src/main/res/menu/activity_main_drawer.xml:
```diff
@@ -17,6 +17,10 @@
             android:id="@+id/nav_drawer_logout"
             android:title="@string/logout" />
 
+        <item
```
Move above `settings` & `logout` items?

tjclawson replied:
Moved and refactored name to Create Hackathon in commit d32666441ae969c5e7a7ef9d87d3396db1105624

tjclawson committed:
Add toast to show response when posting hackathon

Co-authored-by: 05nelsonm <mjnelso2@protonmail.com>
Co-authored-by: msmaitran <itsmaitran@gmail.com>

05nelsonm commented on app/src/main/java/com/lambdaschool/hackathon_portal/ui/fragments/add/AddHackathonFragment.kt:
```diff
@@ -0,0 +1,65 @@
+        }
+
+        fab_save_hackathon.setOnClickListener {
+            val newHackathon = Hackathon(edit_text_hackathon_name.text.toString(),
```
Add logic to to check if `name`, `start_date`, & `end_date` are empty before submitting a network call.

tjclawson replied:
Implemented in commit a646269149804cc3f463b9271fb2cd0413b974dd

tjclawson committed:
Fix typo in FragmentComponent.kt

tjclawson committed:
Change findNavController to calling navController property directly, remove manually setting title

Co-authored-by: 05nelsonm <mjnelso2@protonmail.com>
Co-authored-by: msmaitran <itsmaitran@gmail.com>

tjclawson committed:
Refactor AddHackathonFragment to CreateHackthonFragment, also refactor all relevant methods and classes to match name

Co-authored-by: 05nelsonm <mjnelso2@protonmail.com>
Co-authored-by: msmaitran <itsmaitran@gmail.com>

tjclawson committed:
Add logic to check if required fields are empty and display error if true when creating hackathon

Co-authored-by: 05nelsonm <mjnelso2@protonmail.com>
Co-authored-by: msmaitran <itsmaitran@gmail.com>

tjclawson committed:
Add logic to display if hackathon is open in text, refactor view ids to be more descriptive
>
> Co-authored-by: 05nelsonm <mjnelso2@protonmail.com>
> Co-authored-by: msmaitran <itsmaitran@gmail.com>

Commit:
relocated and cleaned up HackathonRepository

Commit:
moved fab icon off of border
